app/javascript/packages/countdown/countdown-alert-element.ts

```diff
@@ -3,22 +3,37 @@ import type { CountdownElement } from './countdown-element';
 export class CountdownAlertElement extends HTMLElement {
   connectedCallback() {
     if (this.showAtRemaining) {
-      this.addEventListener('lg:countdown:tick', this.handleCountdownTick);
+      this.addEventListener('lg:countdown:tick', this.handleShowAtRemainingTick);
+    }
+
+    if (this.redirectURL) {
+      this.addEventListener('lg:countdown:tick', this.handleRedirectTick);
     }
   }
 
   get showAtRemaining(): number | null {
     return Number(this.getAttribute('show-at-remaining')) || null;
   }
 
+  get redirectURL(): string | null {
+    return this.getAttribute('redirect-url') || null;
+  }
+
   get countdown(): CountdownElement {
     return this.querySelector('lg-countdown')!;
   }
 
-  handleCountdownTick = () => {
+  handleShowAtRemainingTick = () => {
     if (this.countdown.timeRemaining <= this.showAtRemaining!) {
       this.show();
-      this.removeEventListener('lg:countdown:tick', this.handleCountdownTick);
+      this.removeEventListener('lg:countdown:tick', this.handleShowAtRemainingTick);
+    }
+  };
+
+  handleRedirectTick = () => {
+    if (this.countdown.timeRemaining <= 0) {
+      window.location.href = this.redirectURL || '#';
```

Comment:
I think since we're only adding the event handler when `redirectURL` exists, we could consider using TypeScript's [non-null assertion operator here](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-). Or at the very least we could duplicate the condition so that we don't redirect at all if there's nowhere to redirect to.

```suggestion
      window.location.href = this.redirectURL!;
```